Soda Error Handling and Refactoring.

AEscobarCruz opened the pull request:
first major part of refactoring DC. Should be all we need to do for the most part, but could hold out for more suggestions.

removed uncessary dep.

Updated a few error messages that I have missed in a few resource files. Created a new error for bad request due to content type errors.

adding fix to resource type and expected in router class.

adding fix to resource type and expected in router class: extended to main and service class also.

removed extra space

mhotan commented on coordinator-external/src/main/scala/com/socrata/datacoordinator/external/DataCoordinatorError.scala:
```diff
@@ -0,0 +1,110 @@
+package com.socrata.datacoordinator.external
+
+sealed class DataCoordinatorError
```
Looking at the package declaration and the reference to some of these String Constants I am assuming these Errors are the ones surfaced to external clients?   If they are, is there a reason why the functionality to compose the message is not within the class itself?

AEscobarCruz replied:
Good question, I have thought about this, but it seems that the errors are designed in a way so that they can be added upon as needed in a general way. I would definitely like to change it as you described, but would take some significant time to get us there. I would say this is a good start to that alternative you mentioned.

rjmac replied:
We should never need to change it; it doesn't matter what normalizer we use, just that it's _some_ single one.  The point of it is just to ensure that "á" and "á" always compare equal to the less intelligent sorts of string comparators whether they come in as LATIN_SMALL_LETTER_A_WITH_ACUTE or LATIN_SMALL_LETTER_A and COMBINING_ACUTE_ACCENT.

rjmac replied:
...oops, wrong place for that comment.

mhotan replied:
Ohh I see, I should have taken a closer look.  Thank you for describing the design choice.  Changing this would be more of a design pivot other then an optimization or refactoring.  So I vote it is not worth it if external clients are satisfied.

mhotan commented on coordinator/src/main/scala/com/socrata/datacoordinator/common/util/DatasetIdNormalizer.scala:
```diff
@@ -0,0 +1,26 @@
+
+object DatasetIdNormalizer {
+
+  val normalizationMode: Normalizer.Mode = Normalizer.NFC
```
Do we ever plan to use any other Normalizer other then NFC?  Should we make this configurable?  Should we have a class Hierarchy where there is a ```BaseNormalizer``` and a subclass of ```NFCNormalizer```?

AEscobarCruz replied:
Its definitely unclear to me that we would use any other, and so unclear if we should make it configurable.

rjmac replied:
We should never need to change it; it doesn't matter what normalizer we use, just that it's _some_ single one. The point of it is just to ensure that "á" and "á" always compare equal to the less intelligent sorts of string comparators whether they come in as LATIN_SMALL_LETTER_A_WITH_ACUTE or LATIN_SMALL_LETTER_A and COMBINING_ACUTE_ACCENT.

rjmac replied:
Oh, and ever changing it would actually break things, as this controls the canonicalization things are sent to the database with.

rjmac replied:
Grarh and then I go and _delete_ the wrong comment.  This is just a terrible morning.  Anyway:

We should never need to change it; it doesn't matter what normalizer we use, just that it's _some_ single one. The point of it is just to ensure that "á" and "á" always compare equal to the less intelligent sorts of string comparators whether they come in as LATIN_SMALL_LETTER_A_WITH_ACUTE or LATIN_SMALL_LETTER_A and COMBINING_ACUTE_ACCENT.

mhotan replied:
Roger that is good to know.  Just making sure then it is good as is.

mhotan commented on coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetResource.scala:
```diff
@@ -0,0 +1,219 @@
+                             DatasetResource.datasetContentsFunc => Exporter.Result[Unit],
+                           withMutationScriptResults: (HttpResponse) => HttpResponse,
+                           formatDatasetId: DatasetId => String) extends ErrorHandlingSodaResource(formatDatasetId) {
+  override val log = org.slf4j.LoggerFactory.getLogger(classOf[DatasetResource])
```
When Scala Logging just wont do?

AEscobarCruz replied:
Sounds reasonable, ill take a look.

rjmac replied:
Just be aware that Scala Logging makes it very easy to produce a logger with the wrong name.  It won't happen in this case (because nothing will be extending `DatasetResource`) but `Logging` uses the class of the runtime instance to name the logger, which isn't necessarily the same as the class where the logging occurs, which is what you usually (almost always, really) want.

mhotan replied:
Not worth changing at the moment it is just another wrapper that follows more Scala like idioms.

rjmac replied:
Though if it is easy, it's totally worth doing.  slf4j's native `{}`-interpolation is a pain to read; scala-logging definitely makes your logging statements less noisy.

mhotan commented on coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetResource.scala:
```diff
@@ -0,0 +1,219 @@
+    val onlyColumns = Option(servReq.getParameterValues("c")).map(_.flatMap { c =>
+      norm(c).split(',').map(new UserColumnId(_)) }).map(UserColumnIdSet(_ : _*))
+    val limit = Option(servReq.getParameter("limit")).map { limStr =>
+      try {
```
Is it possible to use Scala's [Try](http://www.scala-lang.org/files/archive/nightly/docs/library/index.html#scala.util.Try) for situations like these?

rjmac replied:
Never use Try.  _Especially_ not in situations like this.

AEscobarCruz replied:
Sounds reasonable, as well as for comprehension should make this manageable.

mhotan replied:
Ok I understand the argument not to use it in this situation but any particular reason not to use ```Try``` in general?

rjmac replied:
It's isomorphic to `Either[Throwable, A]` which is a terrifyingly vague type (`Throwable` is the next worst thing to a straight `AnyRef` in terms of precision of meaning).  What `Try` is for (my "never" before is hyperbole) is to allow vaguely try-catch looking code in a situation where you can't actually use try-catch because your control flow isn't block-oriented and so there is no dynamically-enclosing place that can know how to handle exceptions.  For example, if you're writing future-based code and so the thing dynamically enclosing you is just the future management runtime, not the code you'll ultimately be returning values for.

In particular, it's good to note that a signature like `A => Try[B]` doesn't promise that it won't throw exceptions (only that _some_ exceptions -- and it doesn't say which! -- will be returned as `Failure`s), so you have won neither safety by using it nor information about what kinds of failures can occur.

mhotan commented on coordinator/src/main/scala/com/socrata/datacoordinator/resources/NotFoundDatasetResource.scala:
```diff
@@ -0,0 +1,112 @@
+                  Header("X-SODA2-Truth-Version", dataVersion.toString) ~>
+                  ContentType(JsonContentType) ~>
+                  Stream { w =>
+                    val bw = new BufferedOutputStream(w)
```
Woah are we really using BufferedOutputStream at the service level?  hmm can rojoma json handle arrays or list type collections?

AEscobarCruz replied:
Same reaction i had to this. I believe the answer is yes, will look into it.

rjmac replied:
Of course it can, but we're not building an AST here; we're streaming a result.  We absolutely don't want to ever put `result` in a `JArray` as that would hold on to the whole thing in memory.

In modern rojoma-json, it might be better to use an `ArrayIteratorEncoder` (from `c.r.j.v3.util`) here, but that didn't exist when this code (which is being moved from elsewhere) was written.

mhotan commented:
Most of my questions were around style which Robert answered.  The other set of questions are dreicted around testing.  Is there a good set of Unit tests that can provide some level of Regression?

AEscobarCruz commented:
Good question. Since most of my changes involve moving around service end-points for clarity and maintainability (and one additional error message change) I would expect that we should test this via integration tests. Local tests are all good, but I cannot find any particular one testing endpoints?

cloudbees-pull-request-builder commented:
[data-coordinator #441](https://socrata-oss.ci.cloudbees.com/job/data-coordinator/441/) SUCCESS
This pull request looks good

znep commented on coordinator/src/main/scala/com/socrata/datacoordinator/service/Router.scala:
```diff
@@ -0,0 +1,67 @@
+    Routes(
+      // "If the thing is parsable as a DatasetId, do something with it, otherwise give a
+      // SODA2 not-found response"
+      Route("/dataset/{OptString}", notFoundDatasetResource),
```
This does not appear compatible with the existing functionality, which lists datasets if you hit `/dataset` and returns an error if you hit `/dataset/thisdoesnotexist`.

AEscobarCruz replied:
Yup, just double checked and had expectations reversed, fixed.

znep commented on coordinator/src/main/scala/com/socrata/datacoordinator/resources/NotFoundDatasetResource.scala:
```diff
@@ -0,0 +1,112 @@
+import com.rojoma.simplearm.util._
+import com.socrata.datacoordinator.external.BodyRequestError
+import com.socrata.datacoordinator.id.DatasetId
+import com.socrata.datacoordinator.service.{Service, ProcessCreationReturns}
```
Unused import Service?

znep commented on coordinator/src/main/scala/com/socrata/datacoordinator/service/CoordinatorErrorsAndMetrics.scala:
```diff
@@ -0,0 +1,115 @@
+        req.getReader
+      } catch {
+        case _: UnsupportedEncodingException => return Left(datasetErrorResponse(UnsupportedMediaType,
+          ContentTypeRequestError.UNKNOWN_CHARSET, "content-type" -> JString(req.getContentType.toString)))
```
I know you didn't add this... but req.getContentType already returns a string.

AEscobarCruz replied:
ha, should have caught that, sorry.

znep commented on coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:
```diff
@@ -251,10 +253,28 @@ object Main {
+
+        val notFoundDatasetResource = NotFoundDatasetResource(_: Option[String], common.internalNameFromDatasetId, operations.makeReportTemporaryFile, operations.processCreation,
+          operations.listDatasets, _: (HttpResponse) => HttpResponse, serviceConfig.commandReadLimit)
+        val datasetResource = DatasetResource(_: DatasetId, operations.makeReportTemporaryFile, serviceConfig.commandReadLimit, operations.processMutation, operations.deleteDataset, operations.exporter, _: (HttpResponse) => HttpResponse, common.internalNameFromDatasetId)
```
Some of these lines seem a bit excessively long...

AEscobarCruz replied:
fixed.

znep commented on coordinator/src/main/scala/com/socrata/datacoordinator/service/Router.scala:
```diff
@@ -0,0 +1,67 @@
+
+      Route("/dataset-rollup/{DatasetId}", datasetRollupResource),
+
+      Route("/secondary-manifest/{OptString}", secondaryManifestsResource),
```
Similar compatibility issues as above.

AEscobarCruz replied:
I tested this one out, seems to behave as it should, it returns a 404 with no payload. were you expecting a json?

znep replied:
This:

```
m@m:~$ curl -v data-coordinator.app.marathon.aws-us-west-2-staging.socrata.net/secondary-manifest; echo
*   Trying 10.110.35.34...
* Connected to data-coordinator.app.marathon.aws-us-west-2-staging.socrata.net (10.110.35.34) port 80 (#0)
> GET /secondary-manifest HTTP/1.1
> User-Agent: curl/7.41.0
> Host: data-coordinator.app.marathon.aws-us-west-2-staging.socrata.net
> Accept: */*
> 
< HTTP/1.1 200 OK
< Date: Wed, 14 Oct 2015 17:03:00 GMT
< Content-Type: application/json; charset=UTF-8
< Connection: close
< Server: Jetty(9.2.z-SNAPSHOT)
< 
* Closing connection 0
["spandex","pg1-staging","pg2-staging","pg3-staging"]
```

Versus:

```
m@m:~$ curl -v localhost:6020/secondary-manifest; echo
*   Trying ::1...
* Connected to localhost (::1) port 6020 (#0)
> GET /secondary-manifest HTTP/1.1
> User-Agent: curl/7.41.0
> Host: localhost:6020
> Accept: */*
> 
< HTTP/1.1 404 Not Found
< Date: Wed, 14 Oct 2015 17:03:02 GMT
< Content-Length: 0
< Server: Jetty(9.2.2.v20140723)
< 
* Connection #0 to host localhost left intact
```

znep commented:
I'm not sure what to do about this ... but for the record below are all the style errors that show up in the version with your changes that aren't in the original.  The vast majority are just code you moved around or that changed line numbers and aren't things you added as such so you aren't obligated to fix them all.  

We have been put in an ugly situation by adding all these style errors without having a limited set of rules we can sanely clean up... so I guess all I can say is that if you felt like picking out a few of the more important seeming ones and fixing them that would be a small step.  Or if any spring out as ones you did add.  I'm definitely not suggesting you should fix them all.  

```
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/common/util/DatasetIdNormalizer.scala:10:6: Public method must have explicit type
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/common/util/DatasetIdNormalizer.scala:13:8: Public method must have explicit type
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/common/util/DatasetIdNormalizer.scala:21:11: Lowercase pattern match (surround with ``, or add : Any)
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetResource.scala: File must end with newline character
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetResource.scala:120:6: Cyclomatic complexity of 22 exceeds max of 10
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetResource.scala:120:6: Method is longer than 50 lines
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetResource.scala:131:41: Avoid using return
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetResource.scala:138:41: Avoid using return
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetResource.scala:141:62: The string literal "latest" appears 2 times in the file.
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetResource.scala:145:11: Lowercase pattern match (surround with ``, or add : Any)
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetResource.scala:149:43: Avoid using return
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetResource.scala:152:66: The string literal "true" appears 2 times in the file.
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetResource.scala:156:10: Avoid using return
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetResource.scala:159:16: Avoid using return
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetResource.scala:174:77: No semicolon at end of line
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetResource.scala:176: File line length exceeds 120 characters
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetResource.scala:218: File line length exceeds 120 characters
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetResource.scala:51:15: Public method must have explicit type
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetResource.scala:52:15: Public method must have explicit type
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetResource.scala:53:15: Public method must have explicit type
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetResource.scala:56:6: Use : Unit = for procedures
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetResource.scala:72:25: The string literal "X-SODA2-Truth-Last-Modified" appears 2 times in the file.
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetResource.scala:73:25: The string literal "X-SODA2-Truth-Version" appears 2 times in the file.
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetResource.scala:74:25: The string literal "X-SODA2-Truth-Copy-Number" appears 2 times in the file.
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetResource.scala:80:26: Use : Unit = for procedures
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetResource.scala:81:24: If block needs braces
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetRollupResource.scala: File must end with newline character
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetRollupResource.scala:14: File line length exceeds 120 characters
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetRollupResource.scala:18:15: Public method must have explicit type
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetSchemaResource.scala:14: File line length exceeds 120 characters
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetSchemaResource.scala:18:15: Public method must have explicit type
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetSecondaryStatusResource.scala:17: File line length exceeds 120 characters
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetSecondaryStatusResource.scala:21:15: Public method must have explicit type
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetSecondaryStatusResource.scala:26:15: Public method must have explicit type
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetSecondaryStatusResource.scala:33:4: If block needs braces
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetSecondaryStatusResource.scala:34:6: Avoid using return
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetSecondaryStatusResource.scala:48:11: Lowercase pattern match (surround with ``, or add : Any)
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetSecondaryStatusResource.scala:49:16: Avoid using return
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/ErrorHandlingSodaResource.scala:6:6: Redundant braces after class definition
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/NotFoundDatasetResource.scala:102:9: Use braces in for comprehensions
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/NotFoundDatasetResource.scala:103:8: If block needs braces
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/NotFoundDatasetResource.scala:35:15: Public method must have explicit type
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/NotFoundDatasetResource.scala:41:15: Public method must have explicit type
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/NotFoundDatasetResource.scala:48:15: Public method must have explicit type
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/NotFoundDatasetResource.scala:56:6: Use : Unit = for procedures
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/NotFoundDatasetResource.scala:68: File line length exceeds 120 characters
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/NotFoundDatasetResource.scala:77:23: Use braces in for comprehensions
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/SecondariesOfDatasetResource.scala:18:15: Public method must have explicit type
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/SecondaryManifestsResource.scala:17:15: Public method must have explicit type
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/SecondaryManifestsResource.scala:29:4: If block needs braces
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/SecondaryManifestsResource.scala:30:6: Avoid using return
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/resources/VersionResource.scala: File must end with newline character
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/secondary/SecondaryLoader.scala:59:21: Avoid using null
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/CoordinatorErrorsAndMetrics.scala:105:48: Avoid using return
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/CoordinatorErrorsAndMetrics.scala:112:6: Public method must have explicit type
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/CoordinatorErrorsAndMetrics.scala:29:6: Public method must have explicit type
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/CoordinatorErrorsAndMetrics.scala:30:4: Imports should be grouped together
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/CoordinatorErrorsAndMetrics.scala:32:6: The string literal "dataset" appears 4 times in the file.
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/CoordinatorErrorsAndMetrics.scala:38:6: The string literal "errorCode" appears 2 times in the file.
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/CoordinatorErrorsAndMetrics.scala:79:6: Public method must have explicit type
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/CoordinatorErrorsAndMetrics.scala:84: File line length exceeds 120 characters
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/CoordinatorErrorsAndMetrics.scala:86:30: Avoid using null
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/CoordinatorErrorsAndMetrics.scala:86:36: Avoid using return
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/CoordinatorErrorsAndMetrics.scala:92:42: Avoid using return
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/CoordinatorErrorsAndMetrics.scala:93:10: The string literal "content-type" appears 3 times in the file.
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/CoordinatorErrorsAndMetrics.scala:97:6: Avoid using return
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:104:6: Public method must have explicit type
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:105:7: Use braces in for comprehensions
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:110:6: The number of parameters should not exceed 8
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:121:7: Use braces in for comprehensions
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:122: File line length exceeds 120 characters
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:130: File line length exceeds 120 characters
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:159:6: Public method must have explicit type
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:173:4: If block needs braces
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:181:6: Cyclomatic complexity of 18 exceeds max of 10
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:181:6: Method is longer than 50 lines
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:181:6: Use : Unit = for procedures
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:186:20: Regular expression matched 'println'
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:194:7: Use braces in for comprehensions
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:209: File line length exceeds 120 characters
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:214: File line length exceeds 120 characters
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:227:12: Insert a space after the start of the comment
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:234:12: Public method must have explicit type
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:246:12: Public method must have explicit type
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:259: File line length exceeds 120 characters
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:262: File line length exceeds 120 characters
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:263: File line length exceeds 120 characters
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:264: File line length exceeds 120 characters
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:282:23: Use : Unit = for procedures
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:294:36: Magic Number
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:300:23: Use : Unit = for procedures
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:315:36: Magic Number
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:34:7: Use braces in for comprehensions
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:354:92: Whitespace at end of line
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:363: File line length exceeds 120 characters
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:365: File line length exceeds 120 characters
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:49:7: Use braces in for comprehensions
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:67:7: Use braces in for comprehensions
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:72:7: Use braces in for comprehensions
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:79:7: Use braces in for comprehensions
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:86:6: Public method must have explicit type
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:87:7: Use braces in for comprehensions
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:92:6: Public method must have explicit type
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:93:7: Use braces in for comprehensions
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Main.scala:99:7: Use braces in for comprehensions
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Router.scala:41:4: Imports should be grouped together
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Service.scala:112:16: The string literal "dataset" appears 13 times in the file.
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Service.scala:128:16: The string literal "type" appears 3 times in the file.
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Service.scala:223:6: Use : Unit = for procedures
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Service.scala:48:72: Magic Number
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Service.scala:56:8: If block needs braces
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Service.scala:66:6: Cyclomatic complexity of 37 exceeds max of 10
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Service.scala:66:6: Method is longer than 50 lines
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Service.scala:79:12: The string literal "column" appears 5 times in the file.
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Service.scala:86:16: The string literal "value" appears 7 times in the file.
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Service.scala:89:16: The string literal "object" appears 2 times in the file.
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/Service.scala:90:16: The string literal "field" appears 2 times in the file.
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/ServiceUtil.scala:15:14: Use : Unit = for procedures
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/ServiceUtil.scala:20:15: Public method must have explicit type
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/ServiceUtil.scala:23:11: Lowercase pattern match (surround with ``, or add : Any)
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/ServiceUtil.scala:29:11: Lowercase pattern match (surround with ``, or add : Any)
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/ServiceUtil.scala:32:6: Use : Unit = for procedures
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/ServiceUtil.scala:36:6: Use : Unit = for procedures
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/ServiceUtil.scala:47:6: Public method must have explicit type
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/ServiceUtil.scala:58:6: Use : Unit = for procedures
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/ServiceUtil.scala:61: File line length exceeds 120 characters
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/ServiceUtil.scala:69:14: Use : Unit = for procedures
+data-coordinator/coordinator/src/main/scala/com/socrata/datacoordinator/service/ServiceUtil.scala:70:12: If block needs braces
```

znep commented on project/Dependencies.scala:
```diff
@@ -39,6 +39,7 @@ object Dependencies {
 
   val typesafeConfig = "com.typesafe" % "config" % versions.typesafeConfig
 
+
```
nit: gratuitous change

znep commented on project/plugins.sbt:
```diff
@@ -2,6 +2,8 @@ resolvers ++= Seq(
   "socrata releases" at "https://repository-socrata-oss.forge.cloudbees.com/release"
 )
 
+
```
nit: gratuitous change

znep commented:
Reviewed up to 57b5d4841849bdb541b710dd3aee7137ae8d717d, see misc comments.

Phew, thats a lot!  Nice to get `Service.scala` pared down.  Definitely some risk of unintended side effects so we will want to ensure we can get a reasonable test pass on it before it goes to production.

AEscobarCruz commented:
So Ill address some of the issues regarding style, but I do want to comment that I do not believe I added a lot of these, for example I do not think that I added 50+ lines to a method that existed and got moved around, (DatasetResource, line 120, method doExportFile).

cloudbees-pull-request-builder commented:
[data-coordinator #447](https://socrata-oss.ci.cloudbees.com/job/data-coordinator/447/) SUCCESS
This pull request looks good

cloudbees-pull-request-builder commented:
[data-coordinator #448](https://socrata-oss.ci.cloudbees.com/job/data-coordinator/448/) SUCCESS
This pull request looks good

cloudbees-pull-request-builder commented:
[data-coordinator #452](https://socrata-oss.ci.cloudbees.com/job/data-coordinator/452/) SUCCESS
This pull request looks good

cloudbees-pull-request-builder commented:
[data-coordinator #453](https://socrata-oss.ci.cloudbees.com/job/data-coordinator/453/) SUCCESS
This pull request looks good

cloudbees-pull-request-builder commented:
[data-coordinator #458](https://socrata-oss.ci.cloudbees.com/job/data-coordinator/458/) SUCCESS
This pull request looks good

cloudbees-pull-request-builder commented:
[data-coordinator #464](https://socrata-oss.ci.cloudbees.com/job/data-coordinator/464/) SUCCESS
This pull request looks good

cloudbees-pull-request-builder commented:
[data-coordinator #466](https://socrata-oss.ci.cloudbees.com/job/data-coordinator/466/) SUCCESS
This pull request looks good

AEscobarCruz commented:
code rebased.

cloudbees-pull-request-builder commented:
[data-coordinator #470](https://socrata-oss.ci.cloudbees.com/job/data-coordinator/470/) SUCCESS
This pull request looks good

znep commented:
:ship:

AEscobarCruz commented:
Thanks, will do.
